### ISSUE TITLE
Maybe spurious ERROR in log in case of decomission abort with consistent topology changes

### ISSUE THREAD

**Issue by nyh** [scylladb/scylladb#17026]
This is a spin-off of https://github.com/scylladb/scylladb/issues/17021.

The dtest `materialized_views_test.py`::`TestMaterializedViews`::`test_mv_resurrected_rows_after_decommission_interrupt`[`without_rbno`] (and also `with_rbno`) fail when consistent topology changes is enabled (tablets are not necessary to see this bug).

This is easily reproduced locally, with the dtest command line, but only after hacking dtest to add consistent-topology-changes:

```
PYTHONPATH=../scylla-ccm pytest --cassandra-dir=$HOME/scylla/build/dev materialized_views_test.py::TestMaterializedViews::test_mv_resurrected_rows_after_decommission_interrupt[without_rbno]
```

(note that with "--tablets", we get a different failure, more on this in another issue - https://github.com/scylladb/scylladb/issues/17027)

This test has an elaborate setup, where it starts with 4 nodes,  one node is decomissioned but then this decomission is interrupted in the middle, after some data was already streamed (the intention is to check that this doesn't cause a mess with materialized views).

With consistent topology changes, this decommission seems to be detected fine, but generates one error message to the log which dtest doesn't like. The log looks like this:
```
INFO  2024-01-28 23:18:48,639 [shard 0:strm] stream_session - [Stream #d3c8ecf1-be22-11ee-9807-02bd16e30e23] Executing streaming plan for Unbootstrap-system_traces-index-4 with peers=127.0.20.4, slave
ERROR 2024-01-28 23:18:48,677 [shard 0:strm] raft_topology - send_raft_topology_cmd(stream_ranges) failed with exception (node state is decommissioning): seastar::rpc::closed_error (connection is closed)
INFO  2024-01-28 23:18:48,677 [shard 0:strm] raft_topology - start rolling back topology change
INFO  2024-01-28 23:18:48,678 [shard 0:strm] raft_topology - rollback a4991d0a-a3ad-4f23-b96b-de822bf0c740 after decommissioning failure to state rollback_to_normal and setting cleanup flag
INFO  2024-01-28 23:18:48,678 [shard 0:strm] raft_topology - updating topology state: rollback a4991d0a-a3ad-4f23-b96b-de822bf0c740 after decommissioning failure to state rollback_to_normal and setting cleanup flag
```

The first INFO is when the test interrupts the decommission. We start rolling back the process but not before an "ERROR" message, and dtest doesn't like this log message. @kbr-scylla is this error message necessary? Can it be downgraded? After all, failures of Raft messages are part of normal Raft operations and the user shouldn't be alerted to them, no?

On one hand, aborting the decommission of node 4 is some sort of cluster-level "error", but on the other hand, it probably doesn't need to generate an "ERROR"-level log message on other nodes (like node 1 here).

**Comment by bhalevy** [comment]
@kostja / @kbr-scylla who should own this issue?

**Comment by kostja** [comment]
@bhalevy There are two issues here: fixing the test and fixing the error. For now the error needs to be suppressed. The error severity will be addressed with due priority (after more serious bugs).
@temichus could you please add a suppression with @aleksbykov ?

**Comment by kbr-scylla** [comment]
I disagree with downgrading the message, it is a failure scenario -- node dies in the middle of decommission -- and thus ERROR on the topology coordinator is appropriate.

I think we should ignore this particular ERROR pattern in this particular test case, as it's expected.

If there are no objections, I'd move the issue to scylla-dtest repo

**Comment by nyh** [comment]
> I disagree with downgrading the message, it is a failure scenario -- node dies in the middle of decommission -- and thus ERROR on the topology coordinator is appropriate.

Maybe, although why should node 1 report it as an error that node 4 is down?

But more importantly, if this node died and came back, why is it a "failure" at all? Just like we don't report a failure when a coordinator can't write to one of the replicas. Isn't it just part of high-availability - a node can come down and then back up, and so on?

Anyway, I don't feel strongly about this and this is your area of expertise, so feel free to make whatever decision that seems right to you.

**Comment by kbr-scylla** [comment]
> I'm moving the issue to scylla-dtest repo then

For some reason I can't do it, it allows some repos but not scylla-dtest

**Comment by nyh** [comment]
I'm working of fixing these MV tests, so I'll do a patch that does what you recommeneded (ignore this error). I don't mind that the issue is in this repository.

**Comment by nyh** [comment]
I can't reproduce this issue any more. Note that since Since commit https://github.com/scylladb/scylla-dtest/commit/a643c4d234033260ef5b5c1f6d0d2b987a69c07c, dtest now enables consistent-topology-changes by default, so running this test should have seen this problem, but it no longer happens.
So closing the issue.
